docs/madr/decisions/056-meshservice-migration.md

```diff
@@ -0,0 +1,140 @@
+# MeshService migration
+
+* Status: accepted
+
+## Context and Problem Statement
+
+This MADR addresses the switch from not having MeshServices at all to MeshServices
+being automatically created, either from `Services` in Kubernetes or from
+`Dataplanes` in Universal.
+
+When upgrading from 2.8 to 2.9, nothing at the data plane level should change.
+But MeshService behaves differently in several different ways.
+
+### Policy
+
+If for example we have a policy targeted at:
+
+```
+spec:
+  targetRef:
+    kind: MeshService
+    name: demo-app_kuma-demo_svc_5000
+```
+
+when switching over to MeshService, this policy will no longer apply.
+
+### Local traffic
+
+Because the `MeshServices` we generate have selectors equivalent to the `Service`,
+they will select the same `Dataplanes` that previously had the `kuma.io/service`
+corresponding to the `Service` and thus, absent any policies, traffic flows to
+the same endpoints and Envoy has the same cluster configuration.
```

Comment:
So I have an app that now talks to `httpbin-0_kuma-demo_svc_8000.mesh:80` - I turn on MeshService generation - does the traffic still work? If so, what do I do next? Do I switch the app to a hostname generated by hostname generator? Same question goes for a service that uses k8s dns.

Reply:
It will work because the old VIP will be generated, that DNS name will point to it and an outbound listener will be added. But this listener will point to a cluster generated "by MeshService" and so is subject to the behavior of MeshService i.e. routes to local zones only atm. The intention is yes, you would switch to the MeshService native name. Same goes for k8s. The listener for the cluster IP also points to the cluster.

I'll add the above to the MADR